Assets/Puppet/PuppetCrashDialogHandler.cs

```diff
@@ -0,0 +1,52 @@
+﻿using AOT;
+using Microsoft.AppCenter.Unity.Crashes;
+using UnityEngine;
+
+public class PuppetCrashDialogHandler : MonoBehaviour
+{
+    public PuppetConfirmationDialog ConfirmationDialog;
+
+    private static object uiLocker = new object();
```

Comment:
Is `Interlocked.CompareExchange` supported here?

Reply:
@kostya-zhidovinov might have more info on this but I think it protects the `shouldShowDialog` field value while dialog is being displayed so if there would be another `UserConfirmationHandler` method call it would wait until the dialog is closed.
Do you think it could be done without locking?

Reply:
@evgeny-pol @MatkovIvan In that case Dialog can be shown once - at the application start, so I guess it could be done without locking.